src/server/serverUtils/statusBarNotifier.ts

```diff
@@ -57,13 +52,31 @@ export class StatusBarNotifier extends Disposable {
 			} else {
 				portsLabel = localize('port', '{0} Ports', this._ports.size);
 			}
-			portsTooltip = localize(
-				'livePreviewRunningOnPort',
-				'Live Preview running on ports: {0}',
-				`\n\t• ${Array.from(this._ports.values()).join('\n\t• ')}`
-			);
 		}
 
+		const bulletPoints: string[] = [];
+
+		this._ports.forEach((port, uriString) => {
+			let workspace;
+			try {
+				workspace = uriString ? vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(uriString)) : undefined;
+			} catch {
+				// no op
+			}
+			bulletPoints.push(`\n\t• ${port} (${workspace ? workspace.name : localize('nonWorkspaceFiles','non-workspace files')})`);

```

Comment:
Should this be moved within the try statement? Currently, it's possible for `workspace` to be set to `undefined` in the try statement on purpose, or for `workspace` to stay `undefined` because something errored with `getWorkspaceFolder` or `parse`.